File: LeetCode/1608特殊数组的特征值.cpp
int cmp(const void *a, const void *b) {
    return *(int *)a - *(int *)b;
}

int specialArray(int* nums, int numsSize){
    qsort(nums, numsSize, sizeof(int), cmp);
    for (int i = 0; i < numsSize; i++) {
        if (numsSize - i <= nums[i]) {
            if (i == 0 || (i != 0 && numsSize - i > nums[i - 1])) return numsSize - i;
        }
    }
    return -1;
}

File: LeetCode/1608特殊数组的特征值.go
func specialArray(nums []int) int {
    // sort.Ints(nums)
    sort.Sort(sort.Reverse(sort.IntSlice(nums))) //进行逆排序
    left, right := 1, len(nums)
    for left <= right {
        mid := left + (right - left) / 2
        if nums[mid - 1] >= mid {
            if mid == len(nums) || nums[mid] < mid {
                return mid
            } else {
                left = mid + 1
            }
        } else {
            right = mid - 1
        }
    }
    return -1
}